### DISCUSSION TITLE
Dynamic task mapping identifiers

### DISCUSSION THREAD

**Discussion by arroadie** [apache/airflow#25105] [Ideas]
Dynamic mapped tasks are awesome. It enabled the UI to be much more responsive when using dynamic tasks and helps to avoid a bunch of top level python code!
One thing that I kind of miss though is the possibility to identify a specific instance by something more than their index. Right now if I need to "find" a specific mapped task instance the faster way is to log the object that derive in the instances in the previous step and use it to identify the index of the dynamic mapped task (let me know if there's a better way).
It would be awesome to be able to push a value to the function in order to make that instance identifiable (from the UI or the database).
Hadoop Yarn allows the usage of tags, which are only metadata but are searchable from the interface or the API and I believe something in that format it would be very useful for dynamic mapped tasks.

**Comment by potiuk** [comment]
Yes. That would be a nice improvement.

**Reply by foldvari** [reply]
Is this related to #23020  ?

**Reply by arroadie** [reply]
the way I see it, yes. The idea was to be able to provide identifiers to mapped tasks (or tag them, so you can group/filter by tag), which will first allow the users to easily understand what each task represent and secondly being able to gather information (via the DB, Admin?) on that specific group of tasks across time.
